datacapture/src/test/java/com/google/android/fhir/datacapture/validation/RegexValidatorTest.kt

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.datacapture.validation
+
+import android.os.Build
+import com.google.common.truth.Truth.assertThat
+import java.net.URI
+import java.text.SimpleDateFormat
+import org.hl7.fhir.r4.model.BooleanType
+import org.hl7.fhir.r4.model.DateType
+import org.hl7.fhir.r4.model.DecimalType
+import org.hl7.fhir.r4.model.Extension
+import org.hl7.fhir.r4.model.IntegerType
+import org.hl7.fhir.r4.model.PrimitiveType
+import org.hl7.fhir.r4.model.Quantity
+import org.hl7.fhir.r4.model.Questionnaire
+import org.hl7.fhir.r4.model.QuestionnaireResponse
+import org.hl7.fhir.r4.model.QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent
+import org.hl7.fhir.r4.model.StringType
+import org.hl7.fhir.r4.model.TimeType
+import org.hl7.fhir.r4.model.UriType
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.P])
+class RegexValidatorTest {
+
+  @Test
+  fun boolean_notMatchRegex_shouldReturnInvalidResult() {
+    checkAnswerNotMatchRegex(regex = "true", value = BooleanType(false))
+  }
+
+  @Test
+  fun boolean_matchRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex(regex = "true", value = BooleanType(true))
+  }
+
+  @Test
+  fun decimal_notMatchRegex_shouldReturnInvalidResult() {
+    checkAnswerNotMatchRegex(regex = "[0-9]+\\.[0-9]+", value = DecimalType(31234))
+  }
+
+  @Test
+  fun decimal_matchRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex(regex = "[0-9]+\\.[0-9]+", value = DecimalType(3.1415926535))
+  }
+
+  @Test
+  fun int_notMatchRegex_shouldReturnInvalidResult() {
+    checkAnswerNotMatchRegex(regex = "[0-9]+", value = IntegerType(-1))
+  }
+
+  @Test
+  fun int_matchRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex(regex = "[0-9]+", value = IntegerType(1234567890))
+  }
+
+  @Test
+  fun dateType_notMatchRegex_shouldReturnInvalidResult() {
```

Comment:
```suggestion
  fun date_notMatchRegex_shouldReturnInvalidResult() {
```

to be consistent with the next test case name